src/lib/form-field/form-field.ts

```diff
@@ -344,10 +344,17 @@ export class MatFormField extends _MatFormFieldMixinBase
     return this._hasLabel() || this.appearance === 'legacy' && this._hasPlaceholder();
   }
 
-  /** Determines whether to display hints or errors. */
-  _getDisplayedMessages(): 'error' | 'hint' {
-    return (this._errorChildren && this._errorChildren.length > 0 &&
-        this._control.errorState) ? 'error' : 'hint';
+  /** Determines whether to display hints, errors or none. */
+  _getDisplayedMessages(): 'error' | 'hint' | 'none' {
```

Comment:
you are right, it will work. but I wanted to avoid returning null as it is not the best practice I believe. on the other hand, the value i have used is not bringing more information, either. what do you think?

btw. i have noticed that tests are not passing after sending pull request, so I still have to fix them

Reply:
I think `null` is appropriate here, as it represents the absence of anything to show